design/UsageValidationRules.md

```diff
@@ -0,0 +1,71 @@
+## Definition of "resources being used"
+
+Resources in WebGPU can be used in different ways that are declared at resource creation time and interact with various validation rules.
+
+Buffers have the following usages (and commands inducing that usage):
```

Comment:
where does the `HOST_ACCESS` go? (i.e. for read/writes from CPU)

Reply:
It might be orthogonal and defined in a `storageMode` that can be either `shared` or `private`. Adding it as a usage might help with "type safety though" so not sure.